app/controllers/questionnaires_controller.rb

```diff
@@ -1,5 +1,44 @@
 class QuestionnairesController < ApplicationController
-  def new
-    render nothing: true
+  def show
+    @firm = current_user.firm
+  end
+
+  def update
+    @firm = current_user.firm
+    @firm.update!(firm_params)
```

Comment:
Can we take the bang off and use this in a conditional rather than rescuing the error? Would make it consistent with elsewhere.